pkg/cri/server/sandbox_run.go

```diff
@@ -281,6 +264,68 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		return nil, fmt.Errorf("failed to get sandbox container info: %w", err)
 	}
 
+	if podNetwork {
```

Comment:
At the beginning, I think we can just follow the origin path which is to create netns mounted path at first, because the code will be readable and easy to maintain. Even if the netns mounted path maybe leaky, it is maintainable and easy to cleanup when containerd restart to check there is any unused netns path. 

Just my two cents.

Reply:
> We could also change this to have a predictable, generated name for the ns mount rather than the current random name (generated inside netns.NewNetNS). 

Agree.

> Or another alternative would be to allow runc to create the network namespace as it creates the sandbox container, then use that namespace for the other containers rather than creating one here explicitly. The kernel reference-counts namespaces; without either an active mount or running process the namespace will be automatically removed.

It is like runtime pre-start hook. It has data-race if the pid has been recycled. We have dicussed this in the user namespace issue https://github.com/containerd/containerd/issues/7063. Let's discuss or resolve this in follow-up.